base/uk.ac.stfc.isis.ibex.scriptgenerator/src/uk/ac/stfc/isis/ibex/scriptgenerator/ScriptGeneratorSingleton.java

```diff
@@ -228,6 +232,35 @@ public void setUp() {
 		setActionParameters(configLoader.getParameters());
 	}
 	
+	private Optional<URL> loadUserManualUrl() {
+	    String preferenceProperty = preferenceSupplier.scriptGeneratorManualURL();
+	    
+	    // Loop through all URLs in the preference property
+	    // and return the first one reachable from the user's network
+	    for (String url : preferenceProperty.split(",")) {
```

Comment:
This is good, but should be placed onto a new thread as per Tom's suggestion, otherwise we will hang the gui thread waiting for connections. I would suggest using the CompletableFuture...thenAccept pattern (can be seen in the PythonInterface class) with a different thread. 

This will mean that you also have to change the greying out of the manual button mechanism. I would suggest by default you have it greyed out, but place a property change listener that listens for when there is a url available which is updated in the thenAccept part of the completable future.